src/sentry/api/endpoints/integrations/sentry_apps/internal_app_token/details.py

```diff
@@ -70,3 +71,28 @@ def delete(self, request: Request, sentry_app, api_token) -> Response:
         )
 
         return Response(status=204)
+
+
+# TODO: when the above endpoint is deprecated, simply move the convert method.
+# This endpoint handler utilizes the id instead.
+@control_silo_endpoint
+class NewSentryInternalAppTokenDetailsEndpoint(SentryInternalAppTokenDetailsEndpoint):
+    owner = ApiOwner.INTEGRATIONS
+    publish_status = {
+        "DELETE": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    # The api_token is actually the id of the api token, but needs the same signature of the parent class method
+    def convert_args(self, request: Request, sentry_app_slug, api_token, *args, **kwargs):
+        try:
+            api_token_obj = ApiToken.objects.get(id=api_token)
```

Comment:
rather than re-using the `api_token` param, can we rename the route param to `token_id`?
or add a TODO: to change the route param once the UI has been updated?

Reply:
In Python we can't have the signatures for childs be different from the parent, which is why I added the comment above:
```
    # The api_token is actually the id of the api token, but needs the same signature of the parent class method
```

In the follow up PR, I'll change it to `api_token_id` when we've switched over the FE